### PR TITLE
[TransferEngine][feature] support multiple metadata servers

### PR DESCRIPTION
**Pull request by alogfans**
In addition to etcd, Mooncake Transfer Engine now supports redis to keep metadata. To use this, you should install `libhiredis-dev` first and compile Mooncake with `-DUSE_REDIS=ON`. Developers can also inherit the `TransferMetadataImpl` class to implement their own metadata storage logic.

This patch also fix problems in #12 (numa binding) and #20 (missing cudaMemcpy).

### CODE REVIEW

File: mooncake-transfer-engine/src/transport/rdma_transport/worker_pool.cpp [hunk 1]
```diff
@@ -23,6 +23,12 @@
 #include "transport/rdma_transport/rdma_endpoint.h"
 #include "transport/rdma_transport/rdma_transport.h"
 
+#ifdef USE_CUDA
+#include <bits/stdint-uintn.h>
+#include <cuda_runtime.h>
+#include <cufile.h>
```
Comment (wxsms) [inline mooncake-transfer-engine/src/transport/rdma_transport/worker_pool.cpp]: `cufile.h` is not needed maybe?

Comment (alogfans) [reply mooncake-transfer-engine/src/transport/rdma_transport/worker_pool.cpp]: Thanks. We have removed this line.

**Comment by ShangmingCai** [comment]
LGTM. Will test today to check if changing from etcd to Redis affects the vllm integration logic.

**Comment by stmatengss** [comment]
1. Please change the title to "[TransferEngine][feature] support multiple metadata servers"
2. Add documentation about installing hiredis (e.g., ``apt-get install libhiredis-dev``)

**Pull request merged**